feat: Refactor documentation components and payloads

swayam25 opened the pull request:
- Updated `package.json`.
- Removed redundant `DiscordComponent` and emoji utility.
- Replaced instances of `NewDiscordComponent` with `DiscordComponent` across multiple documentation files for consistency.
- Rearranged import statements for better organization in help-command, prefixed-commands, and other related files.
- Updated payloads to include code formatting in help responses.
- Improved button styles in `app.css` for better UI consistency.

swayam25 committed:
Refactor documentation components and payloads

- Updated `package.json`.
- Removed redundant `DiscordComponent` and emoji utility.
- Replaced instances of `NewDiscordComponent` with `DiscordComponent` across multiple documentation files for consistency.
- Rearranged import statements for better organization in help-command, prefixed-commands, and other related files.
- Updated payloads to include code formatting in help responses.
- Improved button styles in `app.css` for better UI consistency.

Paillat-dev commented on src/styles/app.css:
```diff
@@ -44,27 +44,45 @@
     .blurple-btn {
         background-color: var(--color-blurple) !important;
-        padding: 0.5rem 1rem !important;
+        padding: 0.3rem 0.8rem !important;
```
1. Is !important strictly necessary here
2. Can we avoid code repetition here for every single button ?

swayam25 replied:
1. Added `!important` just for safety, making sure that it works as expected.
2. Yes we can avoid code repetition, we can make a separate `btn` class and apply that class to every child class.

Paillat-dev replied:
I get you here, but "for safety" isn't a valid reason. Also, since these are custom classes from what I can see, !important should not be necessary at all. Could you apply the changes to avoid repetition ?

Paillat-dev replied:
From what I understand, those css classes refer to the previous component system and are not needed anymore, right ?

Paillat-dev commented on src/styles/app.css:
```diff
@@ -44,27 +44,45 @@
 
 @layer pycord {
     /* Pycord customizations */
+    button,
```
We should find a way not to make this global, not apply on every single button or a element. Same for the img below. There is probably some class on all starlight images

swayam25 replied:
To improve the visual quality and user experience, we should add transition to every button and hyperlink. The default absence of transitions in Starlight does not look good.

Paillat-dev replied:
@swayam25

swayam25 replied:
Buttons and links need transitions in most cases, so applied them globally

Paillat-dev replied:
They don't need transitions to work, and again, applying styles globally is really not recommended. I doubt that most links look bad w/out transitions, as for buttons they can be targeted more specifically imo, there aren't that many of them and most are in the same locations and so could be targeted by parent or something

swayam25 replied:
So we will remove transition from links?

Paillat-dev replied:
We will *not add them* globally. If you think that for example wiki contents should have hyperlink transitions, feel free to add them, but to that only by targeting them more specifically, and not globally.

swayam25 replied:
Ok

Paillat-dev reviewed (changes requested):
See comments

swayam25 commented on src/styles/app.css:
```diff
@@ -44,27 +44,45 @@
+    }
+
+    img {
+        border: 1px solid var(--color-gray-600) !important;
```
It might not be necessary, but I added `!important` to prevent a Starlight class from overriding the image styles.

Paillat-dev replied:
If it's not necessary, don't do it. And if it is, try avoiding it. Also the same comment as above 100% applies here.

Paillat-dev replied:
This should also not be global. Images are normally present in the wiki contents, and should either have specific classes or target only images inside of some class that is present on the wiki contents

Paillat-dev reviewed (changes requested):
I really don't want to sound pedantic with this, and if I do it was not my intention, but the reason why I am checking the changes like this is because all of these things are good practices  deem required for a project such as this one to be shipped in production *especially* in a project that will implicate multiple developers.

swayam25 commented:
> I really don't want to sound pedantic with this, and if I do it was not my intention, but the reason why I am checking the changes like this is because all of these things are good practices deem required for a project such as this one to be shipped in production _especially_ in a project that will implicate multiple developers.

Uh ok I understand...

swayam25 committed:
Merge branch 'master' into feat/new-theme

swayam25 committed:
Fix css as per suggestions

- Added `font-discord` variable with proper fallback.
- Refactored custom buttons to and minified css.
- Updated `img` class styles to only target `img[data-image-component]`.
- Removed unwanted `!important` declarations.
- Updated images in `/getting-started/creating-your-first-bot`

Paillat-dev commented on package.json:
```diff
@@ -26,8 +26,10 @@
         "svelte": "^5.39.8",
-        "tailwindcss": "^4.1.14"
+        "tailwindcss": "^4.1.14",
+        "typescript": "^5.9.3"
```
@swayam25

swayam25 replied:
Done!

Paillat-dev commented on src/content/docs/getting-started/creating-your-first-bot.mdx:
```diff
@@ -57,10 +57,12 @@ Now, lets get your bot's token. To do this, you want to:
 
-2. Click on the **Copy** button in the "Token" section.
-   ![Token section](https://gblobscdn.gitbook.com/assets%2F-MjPk-Yu4sOq8KGrr_yG%2F-MjdbU12JISJorAZxrKH%2F-MjdbpUsapzb5n15Po5P%2Fimage.png?alt=media&token=118e259f-940a-4f6c-b3a3-c29f3a54100d)
+3. Click on the <button class="blurple-btn">Copy</button> button.
```
If we want this to be only for display we should not use the `button` element because it's gonna get picked up by screendreaders, use a `span` with the necessary styling

swayam25 replied:
Done!

Paillat-dev commented on src/content/docs/index.mdx:
```diff
@@ -14,37 +14,32 @@ hero:
           link: https://github.com/Pycord-Development/pycord
           icon: external
-          variant: minimal
+          variant: secondary
```
What does this change ?

swayam25 replied:
Looks better

swayam25 committed:
Refactor code to use `span` instead of `button` for inline elements

swayam25 committed:
Minor fix

swayam25 committed:
Fix image styling for lazy-loaded images and remove unnecessary styles for hero images
